README.md

```diff
@@ -13,6 +13,32 @@ The Devfile Parser library is a Golang module that:
 3. generates Kubernetes objects for the various devfile resources.
 4. defines util functions for the devfile.
 
+## Private Repository Support
+
+Tokens are required to be set in the following cases:
+1. parsing a devfile from a private repository
+2. parsing a devfile containing a parent devfile from a private repository [1]
+3. parsing a devfile from a private repository containing a parent devfile from a public repository [2]
+
+Set the token for the repository:
+```go
+parser.ParserArgs{
+	...
+	URL: <url-to-devfile-on-supported-git-provider>
+	Token: <repo-personal-access-token>
+	...
+}
+```
+Note: The url must also be set with a supported git provider repo url.
+
+For more information about personal access tokens:
+1. [GitHub docs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+2. [GitLab docs](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token)
+3. [Bitbucket docs](https://support.atlassian.com/bitbucket-cloud/docs/repository-access-tokens/)
+
+[1] Currently, this works under the assumption that the token can authenticate the devfile and the parent devfile; both devfiles are in the same repository.
+[2] In this scenario, the token will be used to authenticate the main devfile.
+
```

Comment:
Can you also update the README to explain the changes that were made in the previous PR (and now expanded by this PR)

https://github.com/devfile/library/pull/139